### PR TITLE
Mobile Post Writing With Multi-select dropdown using Bootstrap

### PR DESCRIPTION
**Pull request by HankG**
related to diaspora/diaspora#7080

### CODE REVIEW

File: app/views/publisher/_publisher.mobile.haml [hunk 1]
```diff
@@ -15,17 +15,16 @@
...
-      - current_user.aspects.each do |aspect|
-        %option{value: aspect.id}
-          = "· #{aspect.name}"
+      .pull-right
+        %select{id: "aspect_ids_", name: "aspect_ids[]", multiple: "multiple"}
+          %option{value: "public"}
+            = t("public")
+          %option{value: "all_aspects", selected: true}
+            = t("all_aspects")
+          %option(data-role="divider")
```
Comment (diaspora-code-review) [inline app/views/publisher/_publisher.mobile.haml]: Prefer the hash attributes syntax (%tag{ lang: 'en' }) over HTML attributes syntax (%tag(lang=en))

Comment (HankG) [reply app/views/publisher/_publisher.mobile.haml]: I haven't figured out how to get this to work with the requested syntax.  It seems that in an option block HAML is only allowing very specific fields.  If someone knows how to fix this I'll implement that. I need this field to create the divider.  Alternatively we could take the divider out if we don't want to ignore the style problem and don't care about the divider.

**Comment by HankG** [comment]
Here are the screenshots on iOS and Android.  Sorry for the Android being a screen-of-a-screen.  I have to get a data transfer process off the Android device I scrounged.  I figured something was better than nothing though.  I've confirmed that the scroll behavior is as expected with a touch screen device.  Unfortunately neither draw the scrollbars very prominently unless the user is scrolling.  I don't think there is anything I can do about that though :(
photo nov 03 14 55 29
photo nov 03 15 52 14

**Comment by SuperTux88** [comment]
I wonder why you choose to use `bootstrap-multiselect` instead of the approach which was used in #7602 (which used the same dropdown was we have everywhere else). While it is up to you if you want to continue an existing PR or start from scratch (as goob commented [here](https://github.com/diaspora/diaspora/issues/7080#issuecomment-435159438)), I actually like the old PR better (at least from a users point of view). It has the same look and feel as everywhere else (including the world icon for public or the lock icon for limited, which is missing here).

While your solution here has some "problems":
  * The checkboxes are too low on both android and iOS
  * It uses system checkboxes which don't adapt to the diaspora theme (white checkboxes on the dark theme)
  * When de-selecting an item, it has a white background which is then difficult to read on the dark theme because light-gray text on white background (I only have an android device and only tested that there, I don't know about iOS)
  * Your dropdown is smaller, so more difficult to click on a touch device
  * I don't like the margin on the left of the checkboxes
  * If multiple aspects are selected, the text isn't translated

While some points are only nitpicking, I still think this is a bit of work. While the other PR basically was working and finished and the only complaint was, that [it was a lot of code duplication](https://github.com/diaspora/diaspora/pull/7602#pullrequestreview-102612377), so only about the code (which I didn't even had a look at in this PR), the functionality and look and feel was good.

So maybe it is still worth to have a look again at the other PR and check if you can reduce the code duplication instead of fixing all the other things in this PR?

(Also, sorry that I didn't comment earlier, I didn't have the time to checkout the state of the old PR earlier)

**Comment by HankG** [comment]
@SuperTux88 I looked at the volume of code in the original PR and it seemed like a lot of code changes compared to the Bootstrap approach. The translation problem is because I used a direct string, which I need to fix.  I have some control over the formatting of the controls by using some styling but the styling differences didn't bother me at all.  I'm a daily user of both the desktop and the mobile sites so I'm not speaking of it as someone who is unaccustomed to both systems.

**Comment by HankG** [comment]
Haven't heard a peep so I'll take a look at resurrecting the old PR rather than continue on with this one...

**Pull request closed without merging**